### PR TITLE
add functions readIEA2024 and readIEA_PRICES

### PR DESCRIPTION
**Pull request by Fsioutas**
add functions readIEA2024 and readIEA_PRICES

### CODE REVIEW

File: R/readIEA2024.R [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#' Read in energy balances from International Energy Agency until period 2022.
+#'
+#' @param subtype Type of data that should be read.
+#' @return The read-in data into a magpie object.
+#'
+#' @author Anastasis Giannousakis, Fotis Sioutas
+#'
+#' @examples
+#' \dontrun{
+#' a <- readSource("IEA2024", subtype = "MAINELEC")
```
Comment (giannou) [inline R/readIEA2024.R:12]:
```
> str(a)
A magpie object (package: magclass)
 @ .Data:  num [1:1066716] NA NA NA NA NA NA NA NA NA NA ...
 $ dimnames:List of 3
  ..$ region                                   : chr [1:249] "ABW" "AFG" "AGO" "AIA" ...
  ..$ period                                   : chr [1:63] "y1960" "y1961" "y1962" "y1963" ...
  ..$ model.scenario.variable.unit.product.flow: chr [1:68] "(Missing).(Missing).(Missing).ktoe.HARDCOAL.MAINELEC" "(Missing).(Missing).(Missing).ktoe.BROWN.MAINELEC" "(Missing).(Missing).(Missing).ktoe.ANTCOAL.MAINELEC" "(Missing).(Missing).(Missing).ktoe.COKCOAL.MAINELEC" ...
```
the object returned has obsolete columns, maybe remove them?

File: R/readIEA_Energy_Projections_Balances.R [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#' Read in Energy_Projections_Balances from International Energy Agency.
+#'
+#' @param subtype Type of data that should be read.
+#' @return The read-in data into a magpie object.
+#'
+#' @author Anastasis Giannousakis, Fotis Sioutas
+#'
+#' @examples
+#' \dontrun{
+#' a <- readSource("IEA_Energy_Projections_Balances", subtype = "INDPROD")
```
Comment (giannou) [inline R/readIEA_Energy_Projections_Balances.R:12]: same as above comment

File: R/readIEA_Energy_Projections_Extended_Indicators.R [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+#' Read in IEA_Energy_Projections_Extended_Indicators from International Energy Agency.
+#'
+#' @param subtype Type of data that should be read.
+#' @return The read-in data into a magpie object.
+#'
+#' @author Anastasis Giannousakis, Fotis Sioutas
+#'
+#' @examples
+#' \dontrun{
+#' a <- readSource("IEA_Energy_Projections_Extended_Indicators", subtype = "TES")
```
Comment (giannou) [inline R/readIEA_Energy_Projections_Extended_Indicators.R:12]: object returned contains a typo ("varius")
also, unclear from the description what the scenarios and "extended indicators" exactly are

File: R/readIEA_PRICES.R [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+#' Read in energy prices from International Energy Agency.
+#'
+#' @param subtype Type of data that should be read.
+#' @return The read-in data into a magpie object.
+#'
+#' @author Anastasis Giannousakis, Fotis Sioutas
+#'
+#' @examples
+#' \dontrun{
+#' a <- readSource("IEA_PRICES", subtype = "ELECTR")
```
Comment (giannou) [inline R/readIEA_PRICES.R:12]: object returned contains a unit value as column name? ("2015 USD/unit"), and obsolete columns
```> str(a)
A magpie object (package: magclass)
 @ .Data:  num [1:39591] NA NA NA NA NA NA NA NA NA NA ...
 $ dimnames:List of 3
  ..$ region                                     : chr [1:249] "ABW" "AFG" "AGO" "AIA" ...
  ..$ period                                     : chr [1:53] "y1970" "y1971" "y1972" "y1973" ...
  ..$ model.scenario.variable.unit.product.sector: chr [1:3] "(Missing).(Missing).(Missing).2015 USD/unit.ELECTR.RESID" "(Missing).(Missing).(Missing).2015 USD/unit.ELECTR.IND" "(Missing).(Missing).(Missing).2015 USD/unit.ELECTR.COMM"
```

**Review by giannou — CHANGES_REQUESTED** [review]

File: R/readIEA2024.R (hunk 1 above)
Comment (Fsioutas) [reply R/readIEA2024.R:12]: ok

File: R/readIEA_PRICES.R (hunk 4 above)
Comment (Fsioutas) [reply R/readIEA_PRICES.R:12]: ok

File: R/readIEA_Energy_Projections_Extended_Indicators.R (hunk 3 above)
Comment (Fsioutas) [reply R/readIEA_Energy_Projections_Extended_Indicators.R:18]: the unit is in the flow, for example Total energy supply (TJ)
if it is written TES does not have units so the unit column is varius
should I use the long name which has the units?

Comment (giannou) [reply R/readIEA_Energy_Projections_Extended_Indicators.R:18]: yes, using the long name makes more sense

Comment (Fsioutas) [reply R/readIEA_Energy_Projections_Extended_Indicators.R]: ok

**Review by giannou — APPROVED** [review]

**Pull request merged**